gym/spaces/dict.py

```diff
@@ -75,66 +89,68 @@ def __init__(
             seed: Optionally, you can use this argument to seed the RNGs of the spaces that make up the :class:`Dict` space.
             **spaces_kwargs: If ``spaces`` is ``None``, you need to pass the constituent spaces as keyword arguments, as described above.
         """
-        assert (spaces is None) or (
-            not spaces_kwargs
-        ), "Use either Dict(spaces=dict(...)) or Dict(foo=x, bar=z)"
-
-        if spaces is None:
-            spaces = spaces_kwargs
+        # Convert the spaces into an OrderedDict
         if isinstance(spaces, dict) and not isinstance(spaces, OrderedDict):
             try:
                 spaces = OrderedDict(sorted(spaces.items()))
             except TypeError:  # raise when sort by different types of keys
                 spaces = OrderedDict(spaces.items())
-        if isinstance(spaces, Sequence):
+        elif isinstance(spaces, Sequence):
             spaces = OrderedDict(spaces)
-
-        assert isinstance(spaces, OrderedDict), "spaces must be a dictionary"
+        elif spaces is None:
+            spaces = OrderedDict()
+        else:
+            assert isinstance(
+                spaces, OrderedDict
+            ), f"Unexpected Dict space input, expecting dict, OrderedDict or Sequence, actual type: {type(spaces)}"
+
+        # Add kwargs to spaces to allow both dictionary and keywords to be used
+        for key, space in spaces_kwargs.items():
+            if key not in spaces:
+                spaces[key] = space
+            else:
+                raise ValueError(
+                    f"Dict space keyword '{key}' already exists in the spaces dictionary."
+                )
 
         self.spaces = spaces
-        for space in spaces.values():
+        for key, space in self.spaces.items():
             assert isinstance(
                 space, Space
-            ), "Values of the dict should be instances of gym.Space"
+            ), f"Dict space element is not an instance of Space: key='{key}', space={space}"
+
         super().__init__(
             None, None, seed  # type: ignore
         )  # None for shape and dtype, since it'll require special handling
 
-    def seed(self, seed: Optional[Union[dict, int]] = None) -> list:
+    def seed(
+        self, seed: Optional[Union[TypingDict[str, int], int]] = None
```

Comment:
I don't think `TypingDict[str, int]` is correct, since we can have nested dictionaries. A recursive type would work, but not in python, so I'd stick with the previous version

Reply:
Good point, I didn't consider that case, I will change back